File: Heaven-Engine/src/gameObjs/gameObject.cpp
#include <gameObjs/gameObject.hpp>

GameObject::GameObject(int sprt, glm::vec2 pos, glm::vec2 siz, glm::vec4 col, bool inter) 
: position(pos), size(siz), color(col), textureIndex(sprt), rotation(0.0f), tag("default"), renderType(RenderType::Default), interpolation(inter) {}

//Define getter functions
bool GameObject::getInterpolationFlag(){
    return this->interpolation;
}

File: heaven.engine/src/main/java/engine/GameObject.java
package engine;

import org.joml.*;
public class GameObject {
    public Vector2f position, size;
    public Vector4f color;
    public float rotation;
    public int textureIndex;

    public GameObject(int texture, Vector2f pos, Vector2f siz, Vector4f clr){
        textureIndex = texture;
        position = pos;
        size = siz;
        color = clr;
        rotation = 0.0f;
    }
}
